### PR TITLE
loading component

### LINKED ISSUE #179: Loading component
Create a Loading component similar to [this one](https://github.com/android/architecture-samples/blob/main/app/src/main/java/com/example/android/architecture/blueprints/todoapp/util/ComposeUtils.kt)

### PR DESCRIPTION
**Pull request by mathieumognol**
Close #179 
Capture d'écran 2024-04-19 203848

### CODE REVIEW

File: app/src/main/java/ch/epfl/skysync/components/Loading.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+fun LoadingComponent(
+    isLoading: Boolean,
+    modifier: Modifier = Modifier,
+    content: @Composable () -> Unit
+) {
+  // The SwipeRefresh component is used to wrap the content. It provides pull-to-refresh
+  // functionality.
+  SwipeRefresh(
+      state = rememberSwipeRefreshState(isLoading),
+      onRefresh = { /* Define what happens on refresh here, e.g., viewModel.loadData()*/},
```
Comment (Plouc314) [inline app/src/main/java/ch/epfl/skysync/components/Loading.kt:37]: Missing a `onRefresh` callback, that the `LoadingComponent` should take

**Review by Plouc314 — COMMENTED** [review]
The general idea looks right, however in the screenshot you provided the `CircularProgressIndicator` is displayed above the content but not on top of it (like in a mille-feuille) is that on purpose ?

**Comment by mathieumognol** [comment]
Thanks for pointing these out, both should be fixed in the last pr

**Review by Plouc314 — APPROVED** [review]
LGTM

**Pull request merged**